### PR TITLE
Job_Template package into nvflare wheel

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description
This PR involves the followings changes: 

1. remove --script argument
2. update tutorials and examples ( for -s removal)
3. remove __init__.py, python_cache
4. setup.py to package job_templates to nvflare/tool/job/templates

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/tutorials/job_cli.ipynb [hunk 1]
```diff
@@ -56,38 +56,17 @@
...
-    "Where the option `-d \"<job_templates_dir>\"` or `--job_template_dir \"<job_templates_dir>\"` is the location of the job_templates. \n",
+    "To list the available templates, you can use the ```nvflare job list_templates``` command:\n",
     "\n",
-    "You can also use the command without the `-d` option:\n",
     "```\n",
     "! nvflare job list_templates\n",
     "```\n",
     "\n",
-    "When the job templates directory is not specified, the Job CLI will try to find the location with the following logic:\n",
+    "If you installed nvflare 2.4.x via `pip install nvflare`. The above command should show you available job templates (built-in default job templates). But if simply clone the github report of repository. The command may ask you to providejob_templates directory. When the job templates directory is not specified, the Job CLI will try to find the location with the following sequences:\n",
```
Comment (IsaacYangSLA) [inline examples/tutorials/job_cli.ipynb:69]: `providejob_templates` should read `provide job_templates`.

**Review by IsaacYangSLA — APPROVED** [review]
Thanks.  Just one small typo.

File: nvflare/tool/job/job_cli.py [hunk 2]
```diff
@@ -183,12 +176,25 @@ def get_src_template(cmd_args) -> Optional[str]:
...
+            file_path = os.path.join(root, f)
+            if not f.endswith(".py") and os.path.exists(file_path):
+                os.remove(file_path)
+        for d in dirs:
+            if d == "__pycache__" or d.endswith(".pyc"):
+                shutil.rmtree(d)
+
+
+def remove_extra_files(config_dir):
+    extra_file = [JOB_INFO_MD, JOB_INFO_CONF, "__init__.py", "__pycache__"]
```
Comment (YuanTingHsieh) [inline nvflare/tool/job/job_cli.py:191]: Not 100% sure if `__init__.py` is extra file though.
Sometimes people do some import there?

File: nvflare/utils/cli_utils.py [hunk 3]
```diff
@@ -196,6 +197,14 @@ def check_job_templates_dir(job_temp_dir: str):
     if job_templates_dir:
         check_job_templates_dir(job_templates_dir)
 
+    if not job_templates_dir:
+        # get default template from nvflare wheel if package installed
+        job_module, import_job_ok = optional_import("nvflare.tool", name="job")
```
Comment (YuanTingHsieh) [inline nvflare/utils/cli_utils.py:202]: A bit confused of this line, when will "import_job_ok" be False?
I thought the nvflare.tool.job is always included in our wheel?

**Review by YuanTingHsieh — COMMENTED** [review]
Good changes, make things more clear, some questions

**Review by IsaacYangSLA — APPROVED** [review]
LGTM.

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM

**Pull request merged**